livekit-agents/livekit/agents/transcription/forwarder.py

```diff
@@ -0,0 +1,241 @@
+            if not segment:
+                return
+            if segment.final:
+                # reset the current id for the next segment
+                self._current_id = segment_uuid()
+            self._event_ch.send_nowait(segment)
+        else:
+            raise ValueError(f"Unknown event type: {type(ev)}")
+
+    async def aclose(self) -> None:
+        """Close the forwarder and cleanup resources."""
+        self._event_ch.close()
+        await aio.cancel_and_wait(self._main_task)
+
+
+class TranscriptionRoomForwarder(TranscriptionForwarder):
+    """Forwards transcriptions to LiveKit rooms."""
+
+    def __init__(
+        self,
+        room: rtc.Room,
+        participant: rtc.Participant | str,
+        track: rtc.Track | rtc.TrackPublication | str | None = None,
+    ):
+        super().__init__()
+        identity = participant if isinstance(participant, str) else participant.identity
+        if track is None:
+            track = find_micro_track_id(room, identity)
+        elif isinstance(track, (rtc.TrackPublication, rtc.Track)):
+            track = track.sid
+
+        self._room = room
+        self._participant_identity = identity
+        self._track_id = track
+
+        self._last_segment_id: Optional[str] = None
+        self._played_text = ""
+
+    @log_exceptions(logger=logger)
+    async def _forward_segment(self, segment: TextSegment) -> None:
+        """Forward transcription segment to LiveKit room."""
+        if not self._room.isconnected():
+            return
+
+        if self._last_segment_id != segment.id:
+            # reset the played text for the new segment
+            self._played_text = ""
+            self._last_segment_id = segment.id
+
+        if segment.is_delta:
+            self._played_text += segment.text
+        else:
+            self._played_text = segment.text
+
+        transcription = rtc.Transcription(
+            participant_identity=self._participant_identity,
+            track_sid=self._track_id,
+            segments=[segment.to_rtc_segment(text=self._played_text)],
+        )
+
+        try:
+            await self._room.local_participant.publish_transcription(transcription)
+        except Exception:
+            logger.exception("Failed to publish transcription")
+
+
+class TranscriptionStreamForwarder(TranscriptionForwarder):
+    """Forwards transcriptions to text streams."""
+
+    def __init__(self, stream: TextIO = sys.stdout):
+        super().__init__()
+        self._stream = stream
+
+    @log_exceptions(logger=logger)
+    async def _forward_segment(self, segment: TextSegment) -> None:
+        """Forward transcription segment to the stream with real-time display."""
+        self._stream.write(segment.text)
+        self._stream.flush()
+
+        if segment.final:
+            self._stream.write("\n")
+            self._stream.flush()
+
+
+DEFAULT_TRANSCRIPTION_TOPIC = "lk.transcription"
```

Comment:
should we change this to "lk.chat" instead or do we want an extra topic just for transcriptions?